client/src/rubin/nublado/client/nubladoclient.py

```diff
@@ -856,6 +860,10 @@ async def auth_to_hub(self) -> None:
         JupyterProtocolError
             Raised if no ``_xsrf`` cookie was set in the reply from the lab.
         """
+        # Remove hub xsrf token.
+        self._hub_xsrf = None
+        # Reinitialize HTTP client
```

Comment:
The comment, or somewhere, should say why this is needed so that we can remember all of this debugging. I'd probably add a note section to the docstring with a pretty comprehensive explanation of what we figured out.